modules/media/php/media.class.inc

```diff
@@ -118,52 +124,32 @@ class Media extends \NDB_Menu_Filter
                );
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @return \Loris\Data\Provisioner
+     */
+    public function getBaseDataProvisioner(): \LORIS\Data\Provisioner
+    {
+        return new MediaFileProvisioner();
+    }
+
     /**
      * Gets the data source for this menu filter.
      *
      * @return \LORIS\Data\Provisioner
      */
-    function getDataProvisioner() : \LORIS\Data\Provisioner
+    public function getDataProvisionerWithFilters() : \LORIS\Data\Provisioner
     {
-        $provisioner = new MediaFileProvisioner();
-
-        $user = \User::singleton();
+        $provisioner = parent::getDataProvisionerWithFilters();
 
-        if ($user->hasPermission("access_all_profiles") == false) {
-            $provisioner = $provisioner->filter(
-                new \LORIS\Data\Filters\UserSiteMatch()
-            );
-        }
-        if (!$user->hasPermission("superuser")) {
+        if (!\NDB_Factory::singleton()->user()->hasPermission('superuser')) {
```

Comment:
since User is already passed to `HideFileFilter->filter` and the filter is only used in this module, I think it would be cleaner to move the permission check there and unconditionally add the filter.

(cc @xlecours since this is basically the same comment I had on one of your PRs and that discussion isn't resolved yet.)